scripts/hid_configurator/led_stream.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+
+import random
+import time
+
+from configurator_core import fetch_free_steps_buffer_info, led_send_single_step
+from configurator_core import Step
+
+MS_PER_SEC = 1000
+
+
+def send_continuous_led_stream(dev, recipient, led_id, freq, substep_cnt = 10):
+    try:
+        # LED stream ends on user request (Ctrl+C) or when an error occurrs.
+        while True:
+            success, (free, ready) = fetch_free_steps_buffer_info(dev, recipient, led_id)
+
+            if not success:
+                return
+
+            if not ready:
+                print('LEDs are not ready')
+                return
+
+            while free > 0:
+                # Send steps with random color and defined duration
+                step = Step(
```

Comment:
the Step class is constructed for each iteration just to store data, provide it to a function where they are 'unpacked' Then they are packted again to other structure and send. Why is that?

Reply:
It was done to pack the data passed to function (reducing number of arguments). Used in similar way as struct in C.

Reply:
To do such "an optimization" of code, you may use dictionary without definition of a class etc. Instead of computing and create of the object in each iteration extend the class with behavior and then use that behavior.

Reply:
I prefer class than dictionary. We limit records, which can be added.

Reply:
There is no limit with a class. If you use class then you have to update three places instead of two.